torch/csrc/jit/runtime/interpreter.cpp

```diff
@@ -701,6 +701,34 @@ struct CodeImpl {
     return r;
   }
 
+  void emitNumInputsOperand(int numInputs) {
+    int result = constant_table_.size();
+    auto ival = IValue(numInputs);
+    // value_to_reg_[ival] = result;
+    constant_table_.emplace_back(std::move(ival));
+    insertInstruction(LOADC, result);
+  }
+
+  void emitTypeCheck(Node* node) {
+    auto num_inputs = node->inputs().size();
+
+    // Check that TypeCheck has at least one input.
+    TORCH_INTERNAL_ASSERT(
+        num_inputs && num_inputs + 1 == node->outputs().size());
+    emitLoadInputs(node->inputs());
+
+    // Since TypeCheck is variadic. Insert an instruction to push number of
+    // input operands into the stack.
+    emitNumInputsOperand(num_inputs);
```

Comment:
I think we could avoid doing this - we are encoding the number of inputs in the instruction itself anyway, probably there is no need in putting that number on stack as well. You could take a look at how `LIST_CONSTRUCT` is emitted for example (`emitContainerConstruct` function).

Reply:
I agree! good catch Mikhail! I had it initially because I was not checking Inst.N.  Fixed